### PR TITLE
add top view for spectating & rm gui item view for this camera mode

### PR DESCRIPTION
**Pull request by luffah**
Hello,  
  following the addition of parameters camera distance and angle in `kart_characteristics.xml` , i observed that soccer camera for spectator mode was too much far. Too, in context of recording video, kim notified a view was missing.

Therefore, the changes are the following :
- add in `user_config.xml`, settings for spectator camera angle and distance 
- add a new camera mode for to have a top view that look the ball in soccer mode,  and look the kart else
- in order to have a full view of the situation, remove the appearance of the player item and speed information when choosing a spectator camera mode

## Agreement
```
By creating a pull request in stk-code, you hereby agree to dual-license your contribution as
GNU General Public License version 3 or any later version and
Mozilla Public License version 2 or any later version.

This includes your previous contribution(s) under the same name of contributor.

Keep the above statement in the pull request comment for agreement.

```

### CODE REVIEW

**Comment by Benau** [comment]
...if you can make it use only the acceleration button to toggle spectate mode so i can add 1 less button in mobile stk spectate screen

**Comment by luffah** [comment]
ok
it is possible

**Comment by luffah** [comment]
Done.

**Comment by Benau** [comment]
```
    if (!isSpectatorCam) 
    if (m_multitouch_gui == NULL || m_multitouch_gui->isSpectatorMode())
    {
        drawPowerupIcons(kart, viewport, scaling);
        drawSpeedEnergyRank(kart, viewport, scaling, dt);
    }
```
indentation

**Comment by luffah** [comment]
ah yes , i let that to make code comparison faster ... coding style shall be (i hope) ok now

File: src/graphics/camera_normal.cpp [hunk 1]
```diff
@@ -344,17 +354,31 @@ void CameraNormal::positionCamera(float dt, float above_kart, float cam_angle,
...
-            Vec3 ball_pos = soccer_world->getBallPosition();
-            Vec3 to_target=(ball_pos-wanted_target);
-            wanted_position = wanted_target + Vec3(0,  fabsf(distance)*tan_up+above_kart, 0) + (to_target.normalize() * distance);
+            SoccerWorld *soccer_world = dynamic_cast<SoccerWorld*> (World::getWorld());
+            if (soccer_world) wanted_target = soccer_world->getBallPosition();
+            wanted_position = wanted_target + Vec3(0,  distance+above_kart, 0);
             m_camera->setPosition(wanted_position.toIrrVector());
             m_camera->setTarget(wanted_target.toIrrVector());
             return;
+            break;
```
[the remark this answers is not in the document]
Comment (luffah) [reply src/graphics/camera_normal.cpp]: One can never be too careful.

[the remark this answers is not in the document]
Comment (luffah) [reply src/graphics/camera_normal.cpp]: Is it a good think for you that i force push ? or do you prefer standard commits ?

[the remark this answers is not in the document]
Comment (Benau) [reply src/graphics/camera_normal.cpp]: Tiny change better force push, btw can you add { } around the indentation change?

[the remark this answers is not in the document]
Comment (luffah) [reply src/graphics/camera_normal.cpp]: https://github.com/supertuxkart/stk-code/pull/4267/files#diff-ed59f4519babb9a8eb34ecd596442e8b

[the remark this answers is not in the document]
Comment (Benau) [reply src/graphics/camera_normal.cpp]: 🐍

**Comment by Benau** [comment]
Btw you missed default: break; in your new code (case)....

**Comment by luffah** [comment]
Damn  , i m too much dependent to code linting tools. And that's when i don't use one...

**Comment by luffah** [comment]
:smile:  i'm preparing my next PR

**Pull request merged**